tests/providers/amazon/aws/utils/test_waiter.py

```diff
@@ -38,6 +38,17 @@ def generate_response(state: str) -> dict[str, Any]:
     }
 
 
+def assert_expected_waiter_type(waiter: mock.MagicMock, expected: str):
+    """
+    There does not appear to be a straight-forward way to assert the type of waiter.
+    Instead, get the class name and check if it contains the expected name.
+
+    :param waiter: A mocked Boto3 Waiter object.
+    :param expected: The expected class name of the Waiter object, for example "ClusterActive".
+    """
+    assert expected in str(type(waiter.call_args[0][0]))
```

Comment:
Cross-PR suggestion

I've checked that specific hook has access to waiters, in my case only custom waiters

https://github.com/apache/airflow/blob/cc8ce585cad5849a561404cf3d2cb726c207a2cb/tests/providers/amazon/aws/waiters/test_custom_waiters.py#L127-L132 


![image](https://user-images.githubusercontent.com/3998685/222123828-b86df7e6-cd32-49d4-a396-4edfbe70653d.png)

Reply:
Not a bad idea.  I wonder if that test should live in the service hook unit tests instead of in test_custom_waiters.py?  So each service hook which presents custom waiters would assert that it is, in fact, presenting those waiters.

Reply:
Good question. I think as more hooks will provide custom waiter as bigger would be `test_custom_waiters.py` so I thought move waiters tests to specific hook would be reasonable

Reply:
I'll add this check in the EMR hook unit tests, I think it's a good suggestion.  May not get to it till tomorrow though.